data_steward/cdr_cleaner/manual_cleaning_rules/dc732_suppress.py

```diff
@@ -0,0 +1,598 @@
+        '-s',
+        '--sandbox_dataset_id',
+        action='store',
+        dest='sandbox_dataset_id',
+        help='Identifies the dataset where output is stored',
+        required=True)
+
+    retract_parser.add_argument(
+        '-c',
+        '--concept_lookup_table',
+        action='store',
+        dest='concept_lookup_table',
+        help='Table where suppressed concepts are loaded',
+        required=True)
+
+    retract_parser.add_argument('-d',
+                                '--dataset_ids',
+                                action='store',
+                                dest='dataset_ids',
+                                nargs='+',
+                                help='Identifies dataset(s) to retract from',
+                                required=True)
+    return parser
+
+
+def setup(concept_lookup_table):
+    """
+    Load BQ tables needed before running
+
+    :param concept_lookup_table: destination table to save lookup
+    """
+    with open(DC732_CONCEPT_LOOKUP_CSV_PATH, 'rb') as csv_file:
+        job_config = bigquery.LoadJobConfig()
+        job_config.source_format = bigquery.SourceFormat.CSV
+        job_config.skip_leading_rows = 1
+        job_config.autodetect = True
+        job = CLIENT.load_table_from_file(csv_file,
+                                          destination=concept_lookup_table,
+                                          job_config=job_config)
+        job.result()
+        print("Loaded {} rows into {}.".format(job.output_rows,
+                                               concept_lookup_table))
+
+
+INVALID_TABLE_REF = (
+    "Invalid table reference: '{table}' "
+    "(qualify with either '<dataset>.<table>' or '<project>.<dataset>.<table>')"
+)
+
+
+def validate_lookup_table_arg(table):
+    """
+    Raise an error if the table reference is not qualified by dataset or project and dataset
+
+    :param table: reference to a table
+    """
+    dot_count = table.count('.')
+    if dot_count < 1 or dot_count > 2:
+        raise ValueError(INVALID_TABLE_REF.format(table=table))
+
+
+def parse_args(args):
+    """
+    Parse command line arguments
+
+    :return: namespace with parsed arguments
+    """
+    parser = get_arg_parser()
+    args = parser.parse_args(args)
+    if args.cmd == 'setup':
+        validate_lookup_table_arg(args.concept_lookup_dest_table)
+    elif args.cmd == 'retract':
+        validate_lookup_table_arg(args.concept_lookup_table)
+    return args
+
+
+if __name__ == '__main__':
+    import sys
+
+    ARGS = parse_args(sys.argv[1:])
+    CLIENT = bq.get_client(ARGS.project_id)
+
+    def get_backup_table_for(table_name):
+        """
+        Helper to get fully qualified table name where backup rows will be stored
+
+        :param table_name: name of table whose rows are to be backed up
+        :return: fully qualified table name
+        """
+        return f'{ARGS.project_id}.{ARGS.sandbox_dataset_id}.{ISSUE_PREFIX}{table_name}'
```

Comment:
why is this function defined here?

Reply:
Hehe I would like your advice on this. I wanted `get_backup_table_for()` to be a simple function to encapsulate the table naming convention and which returns it in fully qualified form (`project_id.dataset_id.table`). The implementation requires `project_id`, `sandbox_dataset_id` which I didn't want to put in the signature of every function. We get those from `ARGS`. Would it be kosher to put the `get_backup_table_for()` definition prior to where `ARGS` is defined? Seems that still wouldn't enable this to be run elsewhere though... Please let me know if I'm just being lazy and should just add `project_id`, `sandbox_dataset_id` to the signature or if you have other ideas. :)

Reply:
Actually, I probably wouldn't make this a function at all because the only thing it is doing is returning a formatted string in two places in one function.  I would pass a fully qualified dataset name to `backup_rows_to_suppress`.  Then I would put a python f-string in `backup_rows_to_suppress` as needed.  

This means you only need to add one parameter to the call to `backup_rows_to_suppress` and you can create it from the arguments available in `retract(...)`.
```
def backup_rows_to_suppress(fq_dataset_name, tables_to_suppress_df, suppress_rows_table):
    # your functions docstring
    tables_group_df = tables_to_suppress_df.groupby(
        by=['table_name', 'ext_table_name'])                                                                                                                                                                                          
    query_jobs = []
    for (table_name, ext_table_name), table_info in tables_group_df:
        dataset_ids = table_info.dataset_id.unique()
        backup_rows_dest_table = f'{fq_dataset_name}.{table_name}'
    # intervening rows
           backup_ext_dest_table = f'{fq_dataset_name}.{ext_table_name}'
```

Reply:
Thanks, that sounds better. Looking at it now...